### ISSUE TITLE
Support multiple connections

### ISSUE THREAD

**Issue by andrewshell** [cadrephp/AtlasOrm.DebugBar.Bridge#1]
@harikt mentioned in a [comment on my blog](https://www.futureproofphp.com/2016/10/14/collecting-data-from-atlas-orm-with-php-debug-bar/#comment-3):

>  Another problem I see is it only used default connection, but if the user has multiple connections, then it will not be handled. May be there need something to deal with the same.

I'm not sure the best way to handle this.  Atlas.ORM can use multiple connections via the `setReadConnection` and `setWriteConnection` methods which pass through to the `setRead` and `setWrite` methods on the `ConnectionLocator`.

They take a name and a callable which is a factory for returning an `ExtendedPdo` object.

This `ExtendedPdo` object will need to be a `Cadre\AtlasOrmDebugBarBridge\ExtendedPdo` because it instanitates a `DebugBar\DataCollector\PDO\TraceablePDO` instead of a plain `PDO` object.

Then the `TraceablePDO` object needs to be added to the `AtlasOrmCollector` object via `addConnection`.

The current way of handling it would be the following, which I feel is messy.

``` php
$atlasContainer = new Cadre\AtlasOrmDebugBarBridge\AtlasContainer(
    'mysql:host=localhost;dbname=testdb',
    'username',
    'password'
);

$debugbar = new DebugBar\StandardDebugBar();
$atlasCollector = new Cadre\AtlasOrmDebugBarBridge\AtlasOrmCollector($atlasContainer);

$atlasContainer->setReadConnection('readonly', function () use ($atlasCollector) {
    $readOnly = new Cadre\AtlasOrmDebugBarBridge\ExtendedPdo(
        'mysql:host=localhost;dbname=slavedb',
        'readonly',
        'password'
    );
    $atlasCollector->addConnection('readonly', $readOnly->getPdo());
    return $readOnly;
});

$debugbar->addCollector($atlasCollector);
```

I'm open to suggestions for how I could make this process cleaner and easier for application developers.

**Comment by pmjones** [comment]
Yeesh, that is messy. And yet after looking at it for a few minutes, I can think of nothing obviously less-messy.

**Comment by harikt** [comment]
Hi all,

I was actually having a different idea. In order to do that probably this need to be resolved https://github.com/atlasphp/Atlas.Orm/issues/6 .

So the idea : 

Use Aura.Sql profiler to retrieve the query and add to the collector. I do understand this again is related to https://github.com/auraphp/Aura.Sql/issues/98 and which will be resolved in 3.x with `psr/log` .

I was actually using https://github.com/fabfuel/prophiler#logging library and you can use the psr-3 log to be logged into that library. But again that is all user preference .

**Comment by harikt** [comment]
This is my understanding, probably for the current time Atlas seems not handling the multiple connection properly . See https://github.com/atlasphp/Atlas.Orm/pull/33

**Comment by andrewshell** [comment]
Do either of you see a problem with this solution? #2 

cc @harikt @pmjones

**Comment by andrewshell** [comment]
``` php
// Configures default connection
$atlasContainer = Cadre\AtlasOrmDebugBarBridge\AtlasContainer(
    'mysql:host=localhost;dbname=testdb',
    'username',
    'password'
);

// Configures readonly connection
$factory = new Cadre\AtlasOrmDebugBarBridge\ConnectionFactory(
    'mysql:host=localhost;dbname=slavedb',
    'readonly',
    'password'
);

// ConnectionFactory is a callable so it works with AtlasContainer
$atlasContainer->setReadConnection('readonly', $factory);

// Creating a new AtlasOrmCollector same as before
$collector = new Cadre\AtlasOrmDebugBarBridge\AtlasOrmCollector($container);

// New method in AtlasOrmCollector that registers itself with the ConnectionFactory
// via setConnection method
$collector->addConnectionFactory($factory, 'readonly');

// The connection from ConnectionFactory doesn't get added to the Collector
// until it is invoked
$debugbar = new DebugBar\StandardDebugBar();
$debugbar->addCollector($collector);
```

**Comment by andrewshell** [comment]
I merged my PR "Introducing ConnectionFactory #2" and tagged it [1.1.0](https://github.com/cadrephp/AtlasOrm.DebugBar.Bridge/tree/1.1.0)
